packages/dashboard/src/components/app.tsx

```diff
@@ -0,0 +1,46 @@
+import classNames from 'classnames';
+import * as React from 'react';
+import { observer } from 'mobx-react-lite';
+import { useDeepEffect } from '@deriv/components';
+import Routes from 'Components/routes/routes';
+import { useStores } from 'Stores';
+import { TUIProps, TClientProps, TConfigProps } from 'Types';
+import './app.scss';
+
+const App: React.FC<TAppProps> = ({ className, client, config, ui, ws }) => {
+    const { client_store, config_store, ui_store } = useStores(ws);
+
+    // Different consumers of this package have different sized headers and footers so they
+    // need to be offset in order to fill the entire page.
+    const dashboard_height = ui.height_offset ? `calc(100vh - ${ui.height_offset})` : '100vh';
+
+    useDeepEffect(() => {
+        ui_store.init(ui);
+        client_store.init(client);
+        config_store.setConfig(config);
+    }, [client, ui]);
```

Comment:
using deep comparison for stores is a costly operation. I reckon we are duplicating stores in `dashboard` package to know the date type of properties in the store.

Instead of duplicating the store files, can we declare types of the stores seperately and use type assertion against the stores that are passed?

we can declare client store interface
```
export interface TClientStore {
    is_logged_in: boolean;
    loginid: string;
    ...
}
```

and then when creating context we can do type assertion
```
React.createContext<TRootStore>({
      client_store: client_store as TClientStore,
      ...
})
```
`client_store` here is from `core` package. Here we do type assertion to tell the compiler about the property types

Reply:
We're not passing full stores from the consumer into the dashboard package, but rather only the values that we would need to operate dashboard (which isn't that much if you check the `TAppProps` type).